ocs_ci/deployment/deployment.py

```diff
@@ -408,97 +408,143 @@ def deploy_ocs_via_operator(self):
             )
 
         # creating StorageCluster
-        cluster_data = templating.load_yaml(constants.STORAGE_CLUSTER_YAML)
+        if self.platform == constants.IBM_POWER_PLATFORM:
+            cluster_data = templating.load_yaml(constants.IBM_STORAGE_CLUSTER_YAML)
+        else:
+            cluster_data = templating.load_yaml(constants.STORAGE_CLUSTER_YAML)
+
         cluster_data['metadata']['name'] = config.ENV_DATA[
             'storage_cluster_name'
         ]
-        deviceset_data = cluster_data['spec']['storageDeviceSets'][0]
-        device_size = int(
-            config.ENV_DATA.get('device_size', defaults.DEVICE_SIZE)
-        )
 
-        # set size of request for storage
-        if self.platform.lower() == constants.BAREMETAL_PLATFORM:
-            pv_size_list = helpers.get_pv_size(storageclass=self.DEFAULT_STORAGECLASS_LSO)
-            pv_size_list.sort()
-            deviceset_data['dataPVCTemplate']['spec']['resources']['requests'][
-                'storage'
-            ] = f"{pv_size_list[0]}"
-        else:
-            deviceset_data['dataPVCTemplate']['spec']['resources']['requests'][
-                'storage'
-            ] = f"{device_size}Gi"
+        if self.platform == constants.IBM_POWER_PLATFORM:
+            deviceset = [None] * 3
 
-        # set storage class to OCS default on current platform
-        if self.DEFAULT_STORAGECLASS:
-            deviceset_data['dataPVCTemplate']['spec'][
-                'storageClassName'
-            ] = self.DEFAULT_STORAGECLASS
+            for i in range(3):
+                deviceset_data = cluster_data['spec']['storageDeviceSets'][i]
+                device_size = int(
+                    config.ENV_DATA.get('device_size', defaults.DEVICE_SIZE)
+                )
 
-        ocs_version = float(config.ENV_DATA['ocs_version'])
+                # set size of request for storage
+                if self.platform.lower() == "powervs":
```

Comment:
I wonder if we should modify the base class deployment method and not create one under the IBM deployment class

Reply:
I will revisit this in the future, I feel there are many platform-specific code in deployment.py under if conditions, they should belong to subclass. But I will look at this restructure in the future with broader discussions.